api/provider/aws/service_create.go

```diff
@@ -1,8 +1,137 @@
 package aws
 
-import "github.com/quintilesims/layer0/common/models"
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
 
 func (s *ServiceProvider) Create(req models.CreateServiceRequest) (*models.Service, error) {
+	var (
+		cluster          string
+		desiredCount     int
+		loadBalancer     *ecs.LoadBalancer
+		loadBalancerRole string
+		serviceName      string
+		taskDefinition   string
+	)
+
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), req.EnvironmentID)
+	cluster = fqEnvironmentID
+
+	desiredCount = 1
+
+	serviceID := generateEntityID(req.ServiceName)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+	serviceName = fqServiceID
+
+	taskDefinitionARN, err := s.lookupTaskDefinitionARNFromDeployID(req.DeployID)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefinition = taskDefinitionARN
+
+	if req.LoadBalancerID != "" {
+		fqLoadBalancerID := addLayer0Prefix(s.Config.Instance(), req.LoadBalancerID)
+		loadBalancerDescription, err := describeLoadBalancer(s.AWS.ELB, fqLoadBalancerID)
+		if err != nil {
+			return nil, err
+		}
+
+		taskDefinitionDescription, err := describeTaskDefinition(s.AWS.ECS, taskDefinitionARN)
+		if err != nil {
+			return nil, err
+		}
+
+		var loadBalancerContainer ecs.LoadBalancer
+		for _, containerDefinition := range taskDefinitionDescription.ContainerDefinitions {
+			for _, portMapping := range containerDefinition.PortMappings {
+				for _, listenerDescription := range loadBalancerDescription.ListenerDescriptions {
```

Comment:
is it possible to break up this nesting of `for` loops?

Reply:
I'm actually not sure we can (I mean, we _can_, but I think it will be way more of a hassle). I had started this comment thinking that maybe we could generate a list of load balancer ports and a list of container ports and then just find a match, but then I realized that we would need the corresponding `containerDefinition.Name`. So I guess we could generate a list of something like
```
type ContainerPort struct {
    Name *string
    Port *int64
}
```
That would save us one level of nesting, and we would still need
```
for _, container := range containerPorts {
    for _, listenerDescription := range loadBalancerDescription.ListenerDescriptions {
        . . .
    }
}
```

I'm happy to entertain alternative ideas if other folks have them, but I don't have one ready at the moment. For reference, [here](https://github.com/quintilesims/layer0/blob/1ce0a511a6566d3e972201fd7c9523042c2b2511/api/backend/ecs/service_manager.go#L234) is the relevant section of the old implementation (this `getLoadBalancer()` function is called by `GetService()` directly above it).